dpytools/validation/json/validation.py

```
@@ -0,0 +1,68 @@
+import json
+from pathlib import Path
+from typing import Dict, Optional, Union
+from urllib.parse import urlparse
+
+import jsonschema
+from jsonschema import ValidationError
+
+"""
+# data_dict is for passing in a dictionary to be validated
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary)
+
+# msg is to include a helpful context when debugging (i.e "what we're validating")
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary, msg="Some helpful message should this validation fail")
+
+validate_json_schema("/path/to/schema.json", data_path="/path/to/some/json", msg="Some helpful message should this validation fail")
+
+# indent should pretty print the json contents of the error to make it
+# more easily parsed by humans
+validate_json_schema("/path/to/schema.json", data_dict=some_dictionary, indent=2)
+"""
+
+
+def validate_json_schema(
+    schema_path: str,
+    data_dict_or_path: Union[Dict, str],
+    msg: str = "",
+    indent: int = 2,
+) -> Optional[ValidationError]:
```

you dont need to return a new error, just raise it for now.